tensorflow/core/lib/gif/gif_io.cc

```diff
@@ -78,11 +78,8 @@ uint8* Decode(const void* srcdata, int datasize,
   if (DGifSlurp(gif_file) != GIF_OK) {
     *error_string = absl::StrCat("failed to slurp gif file: ",
                                  GifErrorStringNonNull(gif_file->Error));
-    // Only stop load if no images are detected.
-    if (gif_file->ImageCount <= 0) {
-      return nullptr;
-    }
-    LOG(WARNING) << *error_string;
+    LOG(ERROR) << *error_string;
```

Comment:
You still have a `gif_file` dereference when building the `error_string`.

Can we make this first check if `gif_file` is `nullptr` and return a standard error string there and then return the existing logic otherwise?

(Unless I'm misunderstanding which ptr is null)

Reply:
`gif_file` is not `nullptr` in this case, it's ok. The `RasterBits` field in `gif_file` is `nullptr` and is dereferenced at `gif_io.cc:178`. 

But maybe we should check `gif_file` for `nullptr` as well.

Reply:
That makes sense, I'll resolve this. Thanks for clarification

Reply:
Actually, can we check the `git_file->RasterBits` field directly?

Reply:
@mihaimaruseac
We can, but the logic of returning error relying on incorrect image counter is still wrong

Reply:
The image counter should be fixed in the dependency, and here we can check both the image counter and the `RasterBits` field until the upstream fix has landed.

Reply:
Ok, I'll try to fix the dependency library and send the PR